searchlib/src/vespa/searchlib/queryeval/dot_product_search.cpp

```diff
@@ -27,7 +27,7 @@ class DotProductSearchImpl : public DotProductSearch
     };
 
     TermFieldMatchData     &_tmd;
-    std::vector<int32_t>    _weights;
+    const std::vector<int32_t> &_weights;
```

Comment:
consider `vespalib::ConstArrayRef<int32_t>`

Reply:
Any particular advantage ?

Reply:
Fewer indirections when accessing the weights.